File: src/0448-Find-All-Numbers-Disappeared-in-Array/find_all_numbers_disappeared_in_array.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    std::vector<int> findDisappearedNumbers(std::vector<int>& nums) {
        std::vector<int> result;

        for (int num : nums) {
            int index = std::abs(num) - 1;
            nums[index] = -std::abs(nums[index]);
        }

        for (int i = 0; i < nums.size(); ++i) {
            if (nums[i] > 0) {
                result.push_back(i + 1);
            }
        }

        return result;
    }
};

int main() {
    std::vector<int> nums = {4,3,2,7,8,2,3,1};
    
    Solution solution;

    std::vector<int> output = solution.findDisappearedNumbers(nums);
    
    for (int i : output) {
        std::cout << i << " ";
    }

    return 0;
}

File: src/0448-Find-All-Numbers-Disappeared-in-Array/find_all_numbers_disappeared_in_array.py
from typing import List

class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        result: List[int] = []

        for num in nums:
            index: int = abs(num) - 1
            nums[index] = -abs(nums[index])
        
        for i, num in enumerate(nums):
            if num > 0:
                result.append(i + 1)
        
        return result

if __name__ == "__main__":
    nums: List[int] = [4,3,2,7,8,2,3,1]
    
    solution = Solution()
    
    print(solution.findDisappearedNumbers(nums))